[AIRFLOW-4456] Add sub-classable BaseBranchOperator (#5231)
mistercrunch/Airflow: docs/concepts.rst

  start_op >> branch_op >> [continue_op, stop_op]

If you wish to implement your own operators with branching functionality,

> you can inherit from ``BranchOperator``, which behaves similarly to

Reviewer: ```suggestion
you can inherit from :class:`~airflow.operators.branch_operator.BaseBranchOperator`, which behaves similarly to
```

> ``BranchPythonOperator`` but expects you to provide an implementation of the

Reviewer: ```suggestion
:class:`~airflow.operators.python_operator.BranchPythonOperator` but expects you to provide an implementation of the
```

> method ``choose_branch``. As with the callable for ``BranchPythonOperator``,

Reviewer: ```suggestion
method ``choose_branch``. As with the callable for BranchPythonOperator,
```
